## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-tls.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#9c26a097-ac03-486c-87fc-5d16697a3ef5) |
| mint-large-bucket.sh | :x: [more...](#38f0467f-e83d-4baa-ac61-4e2a12f8b0a2) |
<a name="38f0467f-e83d-4baa-ac61-4e2a12f8b0a2"></a>
### 6460-19fd597/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      minikube:30310
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 7e03f90a5a20:/mint/log /tmp/mint-logs'
(1/13) Running awscli tests ... done in 55 seconds
(2/13) Running aws-sdk-go tests ... done in 3 seconds
(3/13) Running aws-sdk-java tests ... done in 4 seconds
(4/13) Running aws-sdk-php tests ... done in 58 seconds
(5/13) Running aws-sdk-ruby tests ... done in 20 seconds
(6/13) Running mc tests ... done in 25 seconds
(7/13) Running minio-dotnet tests ... done in 4 minutes and 33 seconds
(8/13) Running minio-go tests ... done in 1 minutes and 24 seconds
(9/13) Running minio-java tests ... done in 2 minutes and 31 seconds
(10/13) Running minio-js tests ... done in 41 seconds
(11/13) Running minio-py tests ... FAILED in 1 hours 4 minutes and 18 seconds
{
  "args": {
    "bucket_name": "minio-py-test-6f58c14f-61b4-4860-b4db-f4996ad7051f",
    "object_name": "d3ad4f8c-9093-47fb-93cf-0276a947b213"
  },
  "status": "FAIL",
  "duration": 3612483,
  "name": "minio-py:test_thread_safe",
  "error": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/connectionpool.py\", line 600, in urlopen\n    chunked=chunked)\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/connectionpool.py\", line 384, in _make_request\n    six.raise_from(e, None)\n  File \"<string>\", line 2, in raise_from\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/connectionpool.py\", line 380, in _make_request\n    httplib_response = conn.getresponse()\n  File \"/usr/lib/python3.5/http/client.py\", line 1197, in getresponse\n    response.begin()\n  File \"/usr/lib/python3.5/http/client.py\", line 297, in begin\n    version, status, reason = self._read_status()\n  File \"/usr/lib/python3.5/http/client.py\", line 266, in _read_status\n    raise RemoteDisconnected(\"Remote end closed connection without\"\nhttp.client.RemoteDisconnected: Remote end closed connection without response\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1431, in test_thread_safe\n    client.remove_object(bucket_name, object_name)\n  File \"/usr/local/lib/python3.5/dist-packages/minio/api.py\", line 971, in remove_object\n    object_name=object_name)\n  File \"/usr/local/lib/python3.5/dist-packages/minio/api.py\", line 1793, in _url_open\n    preload_content=preload_content)\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/poolmanager.py\", line 322, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/connectionpool.py\", line 667, in urlopen\n    **response_kw)\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/connectionpool.py\", line 667, in urlopen\n    **response_kw)\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/connectionpool.py\", line 667, in urlopen\n    **response_kw)\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/connectionpool.py\", line 667, in urlopen\n    **response_kw)\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/connectionpool.py\", line 667, in urlopen\n    **response_kw)\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/connectionpool.py\", line 638, in urlopen\n    _stacktrace=sys.exc_info()[2])\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/util/retry.py\", line 398, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minikube', port=30310): Max retries exceeded with url: /minio-py-test-6f58c14f-61b4-4860-b4db-f4996ad7051f/d3ad4f8c-9093-47fb-93cf-0276a947b213 (Caused by ProtocolError('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',)))\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1845, in main\n    test_thread_safe(client, testfile, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 1434, in test_thread_safe\n    raise Exception(err)\nException: HTTPConnectionPool(host='minikube', port=30310): Max retries exceeded with url: /minio-py-test-6f58c14f-61b4-4860-b4db-f4996ad7051f/d3ad4f8c-9093-47fb-93cf-0276a947b213 (Caused by ProtocolError('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',)))\n",
  "message": "HTTPConnectionPool(host='minikube', port=30310): Max retries exceeded with url: /minio-py-test-6f58c14f-61b4-4860-b4db-f4996ad7051f/d3ad4f8c-9093-47fb-93cf-0276a947b213 (Caused by ProtocolError('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',)))",
  "function": "put_object(bucket_name, object_name, data, length, content_type, metadata, sse)"
}

Executed 10 out of 13 tests successfully.
```
<a name="9c26a097-ac03-486c-87fc-5d16697a3ef5"></a>
### 6460-19fd597/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minikube:30056
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp c35af8a14d45:/mint/log /tmp/mint-logs'
(1/13) Running awscli tests ... done in 2 minutes and 43 seconds
(2/13) Running aws-sdk-go tests ... done in 1 seconds
(3/13) Running aws-sdk-java tests ... done in 2 seconds
(4/13) Running aws-sdk-php tests ... done in 37 seconds
(5/13) Running aws-sdk-ruby tests ... done in 40 seconds
(6/13) Running mc tests ... done in 7 minutes and 10 seconds
(7/13) Running minio-dotnet tests ... FAILED in 5 minutes and 7 seconds

Unhandled Exception: System.AggregateException: One or more errors occurred. (Assert.AreEqual failed. Expected:<0x8D619538BA5DCC4-1>. Actual:<0x8D619538B328CD8-1>. ) ---> Microsoft.VisualStudio.TestTools.UnitTesting.AssertFailedException: Assert.AreEqual failed. Expected:<0x8D619538BA5DCC4-1>. Actual:<0x8D619538B328CD8-1>. 
   at Microsoft.VisualStudio.TestTools.UnitTesting.Assert.HandleFail(String assertionName, String message, Object[] parameters)
   at Microsoft.VisualStudio.TestTools.UnitTesting.Assert.AreEqual[T](T expected, T actual, String message, Object[] parameters)
   at Minio.Functional.Tests.FunctionalTest.CopyObject_Test3(MinioClient minio) in /mint/run/core/minio-dotnet/FunctionalTest.cs:line 960
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at Minio.Functional.Tests.FunctionalTest.Main(String[] args) in /mint/run/core/minio-dotnet/FunctionalTest.cs:line 223

Executed 6 out of 13 tests successfully.
```

--- a/cmd/bitrot.go
+++ b/cmd/bitrot.go
@@ -138,7 +138,9 @@ func (b *bitrotReader) ReadChunk(offset int64, length int64) ([]byte, error) {
 	if b.buf == nil {
 		b.buf = make([]byte, b.endOffset-offset)
 		if _, err := b.disk.ReadFile(b.volume, b.filePath, offset, b.buf, b.verifier); err != nil {
-			logger.LogIf(context.Background(), err)
+			ctx := context.Background()
+			logger.GetReqInfo(ctx).AppendTags("disk", b.disk.String())
+			logger.LogIf(ctx, err)
 			return nil, err
 		}
 	}

--- a/cmd/erasure-encode_test.go
+++ b/cmd/erasure-encode_test.go
@@ -28,6 +28,10 @@ import (
 
 type badDisk struct{ StorageAPI }
 
+func (a badDisk) String() string {
+	return "bad-disk"
+}
+
 func (a badDisk) AppendFile(volume string, path string, buf []byte) error {
 	return errFaultyDisk
 }

--- a/cmd/logger/reqinfo.go
+++ b/cmd/logger/reqinfo.go
@@ -117,6 +117,9 @@ func GetReqInfo(ctx context.Context) *ReqInfo {
 		if ok {
 			return r
 		}
+		r = &ReqInfo{}
+		SetReqInfo(ctx, r)
+		return r
 	}
 	return nil
 }

--- a/cmd/prepare-storage.go
+++ b/cmd/prepare-storage.go
@@ -29,7 +29,7 @@ var printEndpointError = func() func(Endpoint, error) {
 	printOnce := make(map[Endpoint]map[string]bool)
 
 	return func(endpoint Endpoint, err error) {
-		reqInfo := (&logger.ReqInfo{}).AppendTags("endpoint", endpoint.Host)
+		reqInfo := (&logger.ReqInfo{}).AppendTags("endpoint", endpoint.String())
 		ctx := logger.SetReqInfo(context.Background(), reqInfo)
 		m, ok := printOnce[endpoint]
 		if !ok {

--- a/cmd/xl-v1-utils.go
+++ b/cmd/xl-v1-utils.go
@@ -305,6 +305,7 @@ func readXLMeta(ctx context.Context, disk StorageAPI, bucket string, object stri
 	xlMetaBuf, err := disk.ReadAll(bucket, path.Join(object, xlMetaJSONFile))
 	if err != nil {
 		if err != errFileNotFound {
+			logger.GetReqInfo(ctx).AppendTags("disk", disk.String())
 			logger.LogIf(ctx, err)
 		}
 		return xlMetaV1{}, err

--- a/cmd/xl-v1.go
+++ b/cmd/xl-v1.go
@@ -78,7 +78,9 @@ func getDisksInfo(disks []StorageAPI) (disksInfo []DiskInfo, onlineDisks int, of
 		}
 		info, err := storageDisk.DiskInfo()
 		if err != nil {
-			logger.LogIf(context.Background(), err)
+			ctx := context.Background()
+			logger.GetReqInfo(ctx).AppendTags("disk", storageDisk.String())
+			logger.LogIf(ctx, err)
 			if IsErr(err, baseErrs...) {
 				offlineDisks++
 				continue

